src/lib.rs

```diff
@@ -1693,6 +1703,29 @@ mod test {
         Ok(())
     }
 
+    #[test]
+    fn test_total_changes() -> Result<()> {
+        let db = Connection::open_in_memory()?;
+        let sql = "CREATE TABLE foo(x INTEGER PRIMARY KEY, value TEXT default '' NOT NULL,
+                                    desc TEXT default '');
+                   CREATE VIEW foo_bar AS SELECT x, desc FROM foo WHERE value = 'bar';
+                   CREATE TRIGGER INSERT_FOOBAR
+                   INSTEAD OF INSERT
+                   ON foo_bar
+                   BEGIN
+                       INSERT INTO foo VALUES(new.x, 'bar', new.desc);
+                   END;";
+        db.execute_batch(sql)?;
+        let total_changes_before = db.total_changes();
+        let changes = db
+            .prepare("INSERT INTO foo_bar VALUES(null, 'baz');")?
+            .execute([])?;
+        let total_changes_after = db.total_changes();
+        assert_eq!(changes, 0);
+        assert_eq!(total_changes_after - total_changes_before, 1);
```

Comment:
Maybe we should not expose `total_changes` but somehting like this:
https://github.com/xerial/sqlite-jdbc/blob/9a072d3c3e0db7d23bbc2e3f6352f1597fb34997/src/main/java/org/sqlite/jdbc3/JDBC3Statement.java#L133-L139

Reply:
do you want me do to pull something like "execute_large" (or what other name?) that would return total_changes instead of changes?

the thing is although you can access changes() from the value returned by "execute",  fn changes() is still public and available in "Connection".
So if I add an "execute_large", please tell me if I let total_changes public in "Connection".

Reply:
Or the alterntive is that "execute" returns total_changes_after - total_changes_before and that fn changes() remains public as it's a breaking change.

Reply:
Sorry, I just feel like `total_changes` is meaningless for a pooled `Connection`.
Only the difference makes sense.